src/main/java/org/mockbukkit/mockbukkit/attribute/AttributeInstanceMock.java

```diff
@@ -20,111 +23,156 @@
 public class AttributeInstanceMock implements AttributeInstance
 {
 
+	@Getter
 	private final @NotNull Attribute attribute;
+	@Getter
 	private final double defaultValue;
-	private double value;
-	private final @NotNull List<AttributeModifier> modifiers;
+	@Getter
+	@Setter
+	private double baseValue;
+	private final @NotNull List<AttributeModifier> modifiers = new ArrayList<>();
+	private final @NotNull Map<net.kyori.adventure.key.Key, AttributeModifier> modifiersByKey = new HashMap<>();
+	private final @NotNull Map<UUID, AttributeModifier> modifiersByUuid = new HashMap<>();
+	private final @NotNull List<AttributeModifier> transientModifiers = new ArrayList<>();
 
 	/**
 	 * Constructs a new {@link AttributeInstanceMock} for the provided {@link Attribute} and with the specified value.
 	 *
 	 * @param attribute The Attribute this is an instance of.
-	 * @param value     The value of the attribute.
+	 * @param defaultValue     The default value of the attribute.
 	 */
-	public AttributeInstanceMock(@NotNull Attribute attribute, double value)
+	public AttributeInstanceMock(@NotNull Attribute attribute, double defaultValue)
 	{
 		Preconditions.checkNotNull(attribute, "Attribute cannot be null");
 		this.attribute = attribute;
-		this.defaultValue = value;
-		this.value = value;
-		modifiers = new ArrayList<>();
-	}
-
-	@Override
-	public @NotNull Attribute getAttribute()
-	{
-		return attribute;
-	}
-
-	@Override
-	public double getBaseValue()
-	{
-		return value;
-	}
-
-	@Override
-	public void setBaseValue(double value)
-	{
-		this.value = value;
+		this.defaultValue = defaultValue;
+		this.baseValue = defaultValue;
 	}
 
 	@Override
 	public @NotNull Collection<AttributeModifier> getModifiers()
 	{
-		return modifiers;
+		return Collections.unmodifiableCollection(modifiers);
 	}
 
 	@Override
-	public @Nullable AttributeModifier getModifier(@NotNull Key key)
+	public @Nullable AttributeModifier getModifier(@NotNull net.kyori.adventure.key.Key key)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkNotNull(key, "Key cannot be null");
```

Comment:
Just a note that this actually throws an illegal argument exception if the key is null in paper.

Reply:
resolved